li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/FederatedSubscriptionState.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2019 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License").  See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.clients.consumer;
+
+import java.util.Collections;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import org.apache.kafka.common.TopicPartition;
+
+// Current subscription/assignment state at the federated level
+class FederatedSubscriptionState {
+  public enum SubscriptionType {
+    NONE, AUTO_TOPICS, AUTO_PATTERN, USER_ASSIGNED
+  }
+
+  private SubscriptionType _subscriptionType;
+
+  private Set<String> _subscription;
+
+  private Pattern _subscribedPattern;
```

Comment:
do we want to also store the original string ?

Reply:
subscribe() for wildcard subscription directly takes Pattern, not a string.